lib/foreman_remote_execution_core/script_runner.rb

```diff
@@ -167,9 +253,14 @@ def settings
     def run_async(command)
       raise 'Async command already in progress' if @started
       @started = false
+      @user_method.reset
+
       session.open_channel do |channel|
         channel.request_pty
-        channel.on_data { |ch, data| publish_data(data, 'stdout') }
+        channel.on_data do |ch, data|
+          publish_data(data, 'stdout')
+          @user_method.on_data(data, ch)
```

Comment:
With this change the password prompt shows up in the jobs output. Maybe it would be nice if this method consumed the output if it matches?

Reply:
Added filtering out of passwords in job output